gh-skeleton

```diff
@@ -0,0 +1,254 @@
+
+  # If we got here we know we have an authenticated user.
+  log_ok "You are logged into GitHub as ${username}."
+
+  log_info "Creating ${dest_org}/${dest_repo} from skeleton ${src_org}/${src_repo}."
+
+  log_info "Cloning skeleton remote repository to the new local repository."
+  git clone --origin "${src_repo}" git@github.com:"${src_org}/${src_repo}".git "${dest_repo}"
+
+  cd "${dest_repo}"
+
+  log_info "Disabling pushing to the upstream (parent) repository."
+  git remote set-url --push "${src_repo}" no_push
+
+  log_info "Creating the lineage.yml file."
+  cat << END_OF_LINE > .github/lineage.yml
+---
+lineage:
+  skeleton:
+    remote-url: https://github.com/${src_org}/${src_repo}.git
+version: '1'
+END_OF_LINE
+
+  log_info "Staging modified files."
+  git add --verbose .
+
+  log_info "Committing staged files to the develop branch."
+  git commit --message "Add lineage configuration"
+
+  log_info "Creating first-commits branch."
+  git checkout -b first-commits
+
+  log_info "Searching and replacing repository name in source files."
+  find . \( ! -regex '.*/\.git/.*' \) -type f -exec \
+    perl -pi -e "s/${src_repo}/${dest_repo}/g" {} \;
+
+  log_info "Staging modified files."
+  git add --verbose .
+
+  log_info "Committing staged files to the first-commits branch."
+  git commit --message "Rename repository references after clone"
+
+  remote_repo_status="unknown"
+  log_info "Checking for remote repository ${dest_org}/${dest_repo}."
+  remote_repo_name=$(gh repo view --json name --jq ".name" "${dest_org}/${dest_repo}") || :
+  if [[ -n ${remote_repo_name} ]]; then
+    log_ok "${dest_org}/${dest_repo} exists."
+    remote_repo_status="existed"
+  else
+    log_warn "${dest_org}/${dest_repo} does not yet exist."
+    log_info "Attempting to create a new remote repository."
+    # Note that gh will attempt to add the remote origin and will throw an error
+    # if it already exists.  We'll add the origin later if gh fails.
+    remote_repo_status="created"
+    gh repo create --public --confirm "${dest_org}/${dest_repo}" || remote_repo_status="failed"
+  fi
+
+  # Handle the different outcomes of the remote repository.
+  case $remote_repo_status in
+    created)
+      log_ok "The remote repository ${dest_org}/${dest_repo} was successfully created."
+      ;;&
+    existed | failed)
+      log_info "Adding a new remote origin for the this repository."
+      # We can add the remote origin now since gh was unable to do it.
+      git remote add origin git@github.com:"${dest_org}/${dest_repo}".git
+      ;;&
+    created | existed)
+      log_info "Pushing develop and first-commit branches to the remote."
+      git push origin develop first-commits --set-upstream
+      log_info "Opening a new pull request for the first-commits branch."
+      log_info "If prompted, please select ${dest_org}/${dest_repo} as the base repository."
+      gh pr create --title "First commits" --assignee=@me --web
```

Comment:
I believe this should be a draft if we're creating it before any work has been done.

```suggestion
      gh pr create --title "First commits" --assignee=@me --draft --web
```

Reply:
Ah yes, that is excellent.

Reply:
This has been covered previously: https://github.com/cisagov/gh-skeleton/pull/1#discussion_r697752596

![gandalf confused](https://user-images.githubusercontent.com/3229435/131731315-ddf899dd-036f-4b34-b381-4d544ada2de2.gif)

Reply:
I missed when this was shown as impossible in https://github.com/cisagov/gh-skeleton/pull/1#discussion_r697752596, oops!